python/surf/xilinx/_AxiSysMonUltraScale.py

```diff
@@ -496,6 +536,12 @@ def convTemp(dev, var):
         fpValue -= 273.6777
         return round(fpValue,3)
 
+    @staticmethod
+    def convSetTemp(dev, var, value):
+        fpValue = (value + 273.6777)*(4096.0/501.3743)
```

Comment:
Can you add an argument to the class where the FPGA is Utlrascale or Ultrascale+?  This temperature equations are slightly different for the two different FPGA fabrics.

Reply:
Temp conversions updated to support ULTRASCALE or ULTRASCALE_PLUS devices